packages/snip20/src/query.rs

```diff
@@ -0,0 +1,315 @@
+use core::fmt;
+use schemars::JsonSchema;
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+
+use cosmwasm_std::{
+    to_binary, Api, Coin, Extern, HumanAddr, Querier, QueryRequest, StdError, StdResult, Storage,
+    Uint128, WasmQuery,
+};
+
+use secret_toolkit_utils::space_pad;
+
+/// TokenInfo response
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct TokenInfo {
+    pub name: String,
+    pub symbol: String,
+    pub decimals: u8,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub total_supply: Option<Uint128>,
+}
+/// ExchangeRate response
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct ExchangeRate {
+    pub rate: Uint128,
+    pub denom: String,
+}
+/// Allowance response
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct Allowance {
+    pub spender: HumanAddr,
+    pub owner: HumanAddr,
+    pub allowance: Uint128,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub expiration: Option<u64>,
+}
+/// Balance response
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct Balance {
+    pub amount: Uint128,
+}
+/// Transaction data
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct Tx {
+    pub id: u64,
+    pub from: HumanAddr,
+    pub sender: HumanAddr,
+    pub receiver: HumanAddr,
+    pub coins: Coin,
+}
+/// TransferHistory response
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct TransferHistory {
+    pub txs: Vec<Tx>,
+}
+/// Minters response
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct Minters {
+    pub minters: Vec<HumanAddr>,
+}
+
+/// SNIP20 queries
+#[derive(Serialize)]
+#[serde(rename_all = "snake_case")]
+pub enum Snip20QueryMsg<'a> {
+    TokenInfo {},
```

Comment:
I'd name the type `QueryMsg` since people will probably access it as `snip20::QueryMsg`